src/main/java/org/opensearch/knn/index/KNNSettings.java

```diff
@@ -76,6 +76,7 @@ public class KNNSettings {
     public static final String MODEL_INDEX_NUMBER_OF_REPLICAS = "knn.model.index.number_of_replicas";
     public static final String MODEL_CACHE_SIZE_LIMIT = "knn.model.cache.size.limit";
     public static final String ADVANCED_FILTERED_EXACT_SEARCH_THRESHOLD = "index.knn.advanced.filtered_exact_search_threshold";
+    public static final String KNN_FAISS_AVX2_DISABLED = "knn.faiss.avx2.disabled";
```

Comment:
I would call this knn.faiss.avx2.enabled - its typical to avoid negations in naming.

Reply:
But, the purpose of this setting is to disable avx2 explicitly from opensearch.yml something like `plugins.security.disabled`

Reply:
Actually, you are right. Security does it this way - https://opensearch.org/docs/latest/security/configuration/disable/.